src/application/usecases/course/checkScheduleDuplicate.ts

```diff
@@ -17,14 +18,23 @@ import {
 /**
  * Return true if the schedules do not overlap comparing to the schedules of registered courses. Return false otherwise.
  */
-export const checkScheduleDuplicate = ({ courseRepository }: Ports) => async (
+export const checkScheduleDuplicate = (
+  { courseRepository }: Ports,
+  registered?: RegisteredCourse[]
+) => async (
   year: number,
   schedules: Schedule[]
 ): Promise<
   boolean | UnauthorizedError | NetworkError | InternalServerError
 > => {
```

Comment:
先に[アーキテクチャのドキュメント](https://github.com/twin-te/twinte-front/blob/main/docs/architecture.md)を共有しとくべきでした…！

`registered`の引数は以下のようにずらしたほうが現行のアーキテクチャに沿っていると思うのですがいかがでしょうか。

`registered` はビジネスロジックに必要な引数ですが、一方で`{ courseRepository }` は DI に必要な引数です。
なので `registered` は `{ courseRepository }` と並べるよりも `year`, `schedules` と並べたほうがいいなと思いました。

```suggestion
export const checkScheduleDuplicate = (
  { courseRepository }: Ports,
) => async (
  year: number,
  schedules: Schedule[],
  registered?: RegisteredCourse[]
): Promise<
  boolean | UnauthorizedError | NetworkError | InternalServerError
> => {
```